### PR TITLE
CCP integration

### PR DESCRIPTION
**Pull request by corest**
- Bootstrap role extended with fetching repositories with ccp-defined services
- CCP role installs ccp, build images and deploy it on top of the Kubernetes, deployed with kargo playbooks

### CODE REVIEW

File: roles/elastic_search/tasks/main.yaml [hunk 1]
```diff
@@ -9,3 +9,5 @@
 #- include: 130_docker_python_libs.yaml
 
 - include: 150_elastic_search_docker_container.yaml
+
+# - include: 160_elastic_search_postconfig.yaml
```
Comment (akscram) [inline roles/elastic_search/tasks/main.yaml]: Is there a reason to have this line commented?

File: roles/ccp/tasks/main.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+- include: 110_copy_artifacts.yaml
+
+- include: 120_install_ccp.yaml
+
+- include: 130_render_ccp_configuration.yaml
+
+- include: 140_build_oss_images.yaml
```
Comment (akscram) [inline roles/ccp/tasks/main.yaml:7]: Is that possible to make this optional and be able to skip this step through setting some variable like `build_oss_images=False (True by default) in group_vars?

File: roles/ccp/tasks/main.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+
+- include: 150_deploy_oss_images.yaml
```
Comment (akscram) [inline roles/ccp/tasks/main.yaml]: The same as with `build_oss_images`.

File: roles/elastic_search/tasks/main.yaml (hunk 1 above)
Comment (corest) [reply roles/elastic_search/tasks/main.yaml]: After moving from v2 to v5, Elasticsearch excluded from configuration shards and replicas count. Now you need to specify it via curl request for every separate shard. I left it for now before discussing the flow of the indexes creating. 
 - If we are going to define services indexes for elasticsearch during deployment - this tasks file will contain definition of those services
- if services will handle creating indexes by themselves - we will remove this task file.

Comment (akscram) [reply roles/elastic_search/tasks/main.yaml]: We decided to follow the second way, i.e. services are responsible for creating indexes which they use.

File: roles/ccp/templates/ccp.conf.j2 [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  frontend:
+    - devops-portal
+  backend:
+    - ceagle
+    - health
+    - availability
+    - performance
+configs:
+  elasticsearch:
+{% for node in groups['elasticsearch_nodes'] %}
```
Comment (akscram) [inline roles/ccp/templates/ccp.conf.j2:38]: Could you please also set all nodes under the `nodes` key? Like that:

```configs:
elasticsearch:
  nodes:
    - host: <host>
      port: <port>
```

File: roles/ccp/templates/ccp.conf.j2 [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+      git_url: {{ repo.git_url }}
+{% endfor %}
+roles:
+  infra:
+    - etcd
+  frontend:
+    - devops-portal
+  backend:
+    - ceagle
+    - health
```
Comment (akscram) [inline roles/ccp/templates/ccp.conf.j2]: In the current implementation this line have to be replaced by the following two:

```
    - health-api
    - health-job
```

**Review by akscram — APPROVED** [review]

**Pull request merged**